File: 1460-Make-Two-Arrays-Equal-by-Reversing-Subarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) 
    {
        map<int,int>frqTar, frqArr;
        for( int i = 0; i < arr.size(); i++ )
            frqTar[target[i]]++, frqArr[arr[i]]++;

        for( int i = 0; i < arr.size(); i++ )
        {
            if( frqTar[arr[i]] != frqArr[arr[i]] )
                return 0;
        }

        return 1;
    }
};

File: 1460-Make-Two-Arrays-Equal-by-Reversing-Subarrays.go
func canBeEqual(target []int, arr []int) bool{
    frqTar := [1001]int{}
	frqArr := [1001]int{}

	for i:= 0; i < len(arr); i++{
		frqTar[target[i]]++
        frqArr[arr[i]]++
	}

	for i := 0; i < len(arr); i++{
		if frqTar[arr[i]] != frqArr[arr[i]]{ 
            return false
        }
	} 

	return true
}